custom_components/abb_powerone_pvi_sunspec/sensor.py

```diff
@@ -26,28 +39,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry, async_add_d
     _LOGGER.debug("(sensor) Inverter Type (str): %s", hub.data["invtype"])
     _LOGGER.debug("(sensor) Serial#: %s", hub.data["comm_sernum"])
     if hub.data["invtype"] == INVERTER_TYPE[101]:
-        for sensor_info in SENSOR_TYPES_SINGLE_PHASE.values():
-            sensor_data = {
-                "name": sensor_info[0],
-                "key": sensor_info[1],
-                "unit": sensor_info[2],
-                "icon": sensor_info[3],
-                "device_class": sensor_info[4],
-                "state_class": sensor_info[5],
-            }
-            sensors.append(ABBPowerOnePVISunSpecSensor(coordinator, entry, sensor_data))
+        add_sensor_defs(coordinator, entry, sensors, SENSOR_TYPES_SINGLE_PHASE);
     elif hub.data["invtype"] == INVERTER_TYPE[103]:
-        for sensor_info in SENSOR_TYPES_THREE_PHASE.values():
-            sensor_data = {
-                "name": sensor_info[0],
-                "key": sensor_info[1],
-                "unit": sensor_info[2],
-                "icon": sensor_info[3],
-                "device_class": sensor_info[4],
-                "state_class": sensor_info[5],
-            }
-            sensors.append(ABBPowerOnePVISunSpecSensor(coordinator, entry, sensor_data))
+        add_sensor_defs(coordinator, entry, sensors, SENSOR_TYPES_THREE_PHASE)
     async_add_devices(sensors)
```

Comment:
Shouldn't this be the last function before the Return? You prepare the sensors with add_sensor_defs, then add sensors to device with async_add_devices.

Reply:
Yes, I have checked and this call are correct in my homeassistant, I have not corrected on my local repository before commit.  I will commit with the other changes